src/charm.py

```diff
@@ -152,21 +137,14 @@ def _on_register_user_action(self, event: ActionEvent) -> None:
         if not container.can_connect():
             self.unit.status = ops.MaintenanceStatus("Waiting for pebble")
             return
-        registration_shared_secret = synapse.get_registration_shared_secret(container=container)
-        if registration_shared_secret is None:
-            event.fail("registration_shared_secret was not found, please check the logs")
-            return
-        user_data = {
-            "username": event.params["username"],
-            "admin": event.params["admin"],
-        }
         try:
-            user = User(**user_data)
-            synapse.register_user(registration_shared_secret=registration_shared_secret, user=user)
-        except (ValidationError, synapse.SynapseAPIError) as exc:
+            password = actions.register_user(
```

Comment:
Do you think it makes more sense to return the user object? If I had to guess what the `register_user` method returns it would be an object representing the user

Reply:
I agree. Changed.